core/server/proxy/src/main/java/alluxio/proxy/s3/S3Constants.java

```diff
@@ -11,17 +11,27 @@
 
 package alluxio.proxy.s3;
 
+import java.nio.charset.Charset;
+import java.nio.charset.StandardCharsets;
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
  * Constants for S3 northbound API.
  */
 @ThreadSafe
 public final class S3Constants {
+  /* Headers */
   public static final String S3_CONTENT_LENGTH_HEADER = "Content-Length";
   public static final String S3_ETAG_HEADER = "ETAG";
+  public static final String S3_TAGGING_HEADER = "x-amz-tagging";
+  public static final String S3_TAGGING_COUNT_HEADER = "x-amz-tagging-count";
+
   public static final String S3_DATE_FORMAT_REGEXP = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'";
   public static final String S3_STANDARD_STORAGE_CLASS = "STANDARD";
 
+  /* S3 Metadata tagging. */
+  public static final String TAGGING_XATTR_KEY = "s3_tags";
+  public static final Charset TAGGING_CHARSET = StandardCharsets.UTF_8;
```

Comment:
On both client and UFS I think the metadata tags would be represented as plaintext, so the use of a charset is purely to keep it consistent within Alluxio since `xAttr` maps bytes and not Objects.